test/turing.jl

```diff
@@ -0,0 +1,25 @@
+@testset "turing compat" begin
+    k = SqExponentialKernel()
+    y = randn(3)
+    X = randn(3, 1)
+    x = [rand(1) for _ in 1:3]
+    @model GPRegression(y, X) = begin
```

Comment:
Just a note: the `@model` syntax just expects a regular Julia function (and returns it with transformed function body), so you could equally well write
```suggestion
    @model function GPRegression(y, X)
```
The tutorials and documentation (should) use this form now as well IIRC.